lib/tpm2_eventlog.h

```diff
@@ -17,23 +17,25 @@ typedef bool (*EVENT2DATA_CALLBACK)(TCG_EVENT2 const *event, UINT32 type,
                                     void *data);
 typedef bool (*SPECID_CALLBACK)(TCG_EVENT const *event, void *data);
 
+typedef struct {
+    void * data;
+    SPECID_CALLBACK specid_cb;
+    EVENT2_CALLBACK event2hdr_cb;
+    DIGEST2_CALLBACK digest2_cb;
+    EVENT2DATA_CALLBACK event2_cb;
+    uint8_t sha1_pcrs[TPM2_MAX_PCRS][TPM2_SHA1_DIGEST_SIZE];
+    uint8_t sha256_pcrs[TPM2_MAX_PCRS][TPM2_SHA256_DIGEST_SIZE];
```

Comment:
I've updated it for all of the current supported ones.  Something that does dynamic allocation of PCR banks and tracks which are in use could be added, although it would require quite a bit additional code.